### DISCUSSION TITLE
插入数据显示成功后集合为空，查不出数据

### DISCUSSION THREAD

**Discussion by ghjhhyuyuy** [milvus-io/milvus#37934] [Q&A and General discussion]
milvus2.1数据插入成功，但是collection为空，可能的原因是什么？drop_collection的机制是什么，集合会立刻删除吗，如果这时候插入数据是否有插入失败的情况？

**Comment by xiaofan-luan** [comment]
1. 请使用2.3或者2.4最新，2.1不推荐在用了
2. 如何判断collection为空？count不准确，flush一下会刷新
3. drop后会立即删除，数据会一段时间之后清理。插入数据肯定会失败

**Reply by ghjhhyuyuy** [reply]
使用collection.is_empty判断的，这个准吗，2.1是不是没有flush？

**Reply by yhmo** [reply]
python sdk 2.1没有暴露flush接口，但你可以调用num_entities间接触发flush
`print(collection.num_entities)`

drop_collection会立即删除集合，集合被删除之后，你调用insert会报collection不存在的错误。
